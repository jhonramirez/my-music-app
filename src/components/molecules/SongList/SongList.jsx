import { icoFavorites } from "../../../svg";
import { alertMessage, insertDeleteSong } from "../../../utils";
import { Input, Label } from "../..";
import "./SongList.scss";

export const SongList = ({ track, favorite, setMessage, onClick }) => {
  const onsubmit = (song) => {
    insertDeleteSong(song, setMessage);
    alertMessage();
  };

  return (
    <div className="songList" onClick={() => onClick({ ...track })}>
      <div className="songList__img">
        <img src={track.album.images[0].url} alt="album" className="img" />
      </div>
      <div className="songList__infoSection">
        <p className="songList__info">
          <Label name={track.name} className="songList__name" />
          <Label
            name={track.artists.map((artist) => artist.name).join(",")}
            className="songList__singer"
          />
        </p>
        <Input
          type="checkbox"
          id={track.name}
          defaultChecked={favorite}
          onClick={() => onsubmit(track)}
        />
        <Label name={icoFavorites} htmlFor={track.name} />
      </div>
    </div>
  );
};
