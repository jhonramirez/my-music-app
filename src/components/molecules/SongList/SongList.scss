img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.songList {
  display: flex;
  align-items: center;
  font-size: 14px;
  font-weight: 600;
  color: #FFF;
  margin: 10px 0;
  padding: 5px;

  &__img {
    width: 90px;
    min-width: 90px;
    height: 90px;
    border-radius: 0.5em;
    background-color: #848484;
    margin-right: 10px;
    overflow: hidden;
  }
  &__infoSection {
    display: flex;
    justify-content: space-between;
    width: 100%;
  }

  &__info {
    display: flex;
    justify-content: space-between;
    align-items: center;
    width: 50%;
  }

  &__name {
    width: 100%;
    font-size: 20px;
    text-align: start;
  }

  &__singer {
    margin-left:20px;
    text-align: right;
    font-size: 18px;
  }
}

input[type="checkbox"] {
  display: none;
}

input[type="checkbox"] + label {
  width: 6em;
  text-align: center;
  cursor: pointer;
}

input[type="checkbox"]:checked + label svg {
  transition: all 300ms ease-in-out;
  fill: rgb(240, 136, 39);
  transform: scale(1.3);
  -webkit-transform: scale(1.1);
}

label svg {
  width: 30px;
  fill: rgb(167, 167, 167);
}

@media only screen and (max-device-width: 750px) {
  .songList {
    &__info {
      flex-direction: column;
      align-items: flex-start;
    }
    &__name,
    &__singer {
      font-size: 15px;
      text-align: start;
      margin:0px;
    }
  }
}
