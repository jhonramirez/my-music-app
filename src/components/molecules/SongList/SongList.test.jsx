import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import { SongList } from "./SongList";

jest.mock("../../../services/createDB");
jest.mock("../../../utils");

describe("Pruebas en <SongList/>", () => {
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<SongList />);
    expect(container).toMatchSnapshot();
  });

  test("Se debe llamar el onsubmit cuando se hace click en el corazon de favoritos", () => {
    render(
      <SongList
        track={"track"}
        favorite={true}
        setMessage={() => {}}
        onClick={() => {}}
      />
    );
    const favoritesElement = screen.getByRole("checkbox");
    const statusChecked = favoritesElement.checked;
    fireEvent.click(favoritesElement);
    expect(favoritesElement.checked).toBe(!statusChecked);
  });

  test("Se evalua que tenga className 'songList__infoSection'", () => {
    render(
      <SongList
        track={"track"}
        favorite={true}
        setMessage={() => {}}
        onClick={() => {}}
      />
    );
    const articleElement = screen.getAllByRole("article")[1];
    expect(articleElement).toHaveClass("songList__infoSection");
  });

  test("Se evalua que tenga className 'songList__singer'", () => {
    const track = {
      album: {
        images: [{ url: "proof" }],
      },
      name: "proof",
      artists: [{ name: "artist" }],
    };
    render(
      <SongList
        track={track}
        favorite={true}
        setMessage={() => {}}
        onClick={() => {}}
      />
    );
    const articleElement = screen.getByText("artist");
    expect(articleElement).toHaveClass("songList__singer");
  });
});
