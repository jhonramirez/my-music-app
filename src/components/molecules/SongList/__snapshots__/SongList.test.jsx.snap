// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SongList/> Debe hacer match con el snapshot 1`] = `
<div>
  <section
    class="songList"
    data-testid="songList"
  >
    <article
      class="songList__img"
    >
      <img
        alt="album"
        class="img"
      />
    </article>
    <article
      class="songList__infoSection"
    >
      <p
        class="songList__info"
      >
        <label
          class="songList__name"
          data-testid="label"
        />
        <label
          class="songList__singer"
          data-testid="label"
        />
      </p>
      <input
        data-testid="heart-input"
        type="checkbox"
      />
      <label
        data-testid="label"
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 21.35l-1.45-1.32c-5.15-4.67-8.55-7.75-8.55-11.53 0-3.08 2.42-5.5 5.5-5.5 1.74 0 3.41.81 4.5 2.09 1.09-1.28 2.76-2.09 4.5-2.09 3.08 0 5.5 2.42 5.5 5.5 0 3.78-3.4 6.86-8.55 11.54l-1.45 1.31z"
          />
        </svg>
      </label>
    </article>
  </section>
</div>
`;
